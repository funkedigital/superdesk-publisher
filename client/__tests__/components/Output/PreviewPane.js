import React from "react";
import PreviewPane from "../../../components/Output/PreviewPane";
import { render, fireEvent } from "@testing-library/react";
import Store from "../../../components/Output/Store";

jest.mock('moment', () => () => ({ fromNow: () => '1 day ago', format: () => '2020–01–01T12:00:00+00:00' }));

const _package = {
  "id": 641,
  "guid": "urn:newsml:sp-api.superdesk.pro:2019-09-24T16:37:56.756543:e027a313-b6ce-432b-91fa-45556e5ff1a8",
  "headline": "All during this wonderful May that we have been having",
  "byline": null,
  "slugline": null,
  "language": "en",
  "type": "text",
  "located": null,
  "urgency": 0,
  "priority": 5,
  "version": 3,
  "ednote": null,
  "description_text": "\"My dear boy,\" said Lord Henry, smiling, \"anybody can be good in the country. There are no temptations there. That is the reason why people who live out of town are so absolutely uncivilized. Civilization is not by any means an easy thing to attain.",
  "pubstatus": "usable",
  "evolvedfrom": null,
  "source": "Input",
  "firstpublished": "2019-09-24T14:44:16+0000",
  "copyrightnotice": null,
  "copyrightholder": null,
  "body_html": "<p>\"Culture and corruption,\" echoed Dorian. \"I have known something of both. It seems terrible to me now that they should ever be found together. For I have a new ideal, Harry. I am going to alter. I think I have altered.\"</p>\n<p>\"You have not yet told me what your good action was. Or did you say you had done more than one?\" asked his companion as he spilled into his plate a little crimson pyramid of seeded strawberries and, through a perforated, shell-shaped spoon, snowed white sugar upon them....",
  "created_at": "2019-09-24T14:44:30+00:00",
  "updated_at": "2019-10-21T13:15:00+00:00",
  "articles": [],
  "status": "published"
};

describe("Output/PreviewPane", () => {

  it("renders correctly", () => {
    const { container } = render(
      <Store.Provider
        value={{
          publisher: { getPackage: id => _package }
        }}
      >
        <PreviewPane package={_package} close={jest.fn()} />
      </Store.Provider>
    );
    expect(container.firstChild).toMatchSnapshot();

  });

});
