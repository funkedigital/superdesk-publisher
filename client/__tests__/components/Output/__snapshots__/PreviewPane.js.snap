// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output/PreviewPane renders correctly 1`] = `
<div
  class="sd-preview-panel"
  style="z-index: 1;"
>
  <div
    class="side-panel side-panel--shadow-right"
  >
    <div
      class="side-panel__header side-panel__header--border-bottom"
    >
      <div
        class="side-panel__tools"
      >
        <a
          class="icn-btn"
          flow="top"
          sd-tooltip="Close"
        >
          <i
            class="icon-close-small"
          />
        </a>
      </div>
      <h3
        class="side-panel__heading"
      >
        Item preview
      </h3>
    </div>
    <div
      class="side-panel__content"
    >
      <div
        class="side-panel-collapsible-header active"
      >
        <div
          class="side-panel-collapsible-header__fixed"
        >
          <p
            class="sd-text__date-and-author"
          >
            <time
              title="2019-10-21T13:15:00+00:00"
            >
              1 day ago
            </time>
            
          </p>
        </div>
        <div
          class="side-panel-collapsible-header__collapsible"
        >
          <div
            class="form__row form__row--small-padding"
          >
            <div
              class="flex-grid flex-grid--wrap-items flex-grid--small-2"
            >
              <div
                class="flex-grid__item"
              >
                <label
                  class="form-label form-label--light"
                >
                  Page views
                </label>
                <p />
              </div>
              <div
                class="flex-grid__item"
              >
                <label
                  class="form-label form-label--light"
                >
                  Comments
                </label>
                <p>
                  0
                </p>
              </div>
            </div>
          </div>
        </div>
        <button
          class="side-panel-collapsible-header__toggle active"
        >
          <i
            class="icon-chevron-down-thin"
          />
        </button>
      </div>
      <div
        class="side-panel__content-block"
      >
        <h3
          class="side-panel__content-block-heading"
        >
          All during this wonderful May that we have been having
        </h3>
        <div
          class="side-panel__content-block-text sp-imageMaxWidth-wrapper"
        >
          <div
            class="spinner spinner--dark"
            data-reactroot=""
          >
            <div
              class="bounce1"
            />
            <div
              class="bounce2"
            />
            <div
              class="bounce3"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
